api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -488,6 +487,14 @@ message Cluster {
     //
     //   If the header appears multiple times only the first value is used.
     bool use_http_header = 1;
+
+    // The http header to override destination address if :ref:`use_http_header <envoy_v3_api_field_config.cluster.v3.Cluster.OriginalDstLbConfig.use_http_header>`.
+    // is set to true. If no field is specified, :ref:`x-envoy-original-dst-host <config_http_conn_man_headers_x-envoy-original-dst-host>`
+    // will be used.
+    oneof http_header_name {
+      // Use the ``:authority`` header to override original dst address.
+      bool use_http_authority = 2;
```

Comment:
why not just take the header name here?

Reply:
It's kind of following "easy things easy". Especially this follows that semantic of CONNECT.
The `oneof` can be extended to any header name if you or anyone need.

Reply:
I also agree a `StringValue http_header_name` with default value `x-envoy-original-dst-host` would be better here.

Reply:
+1 this seems better to just allow changing the header with a default being the existing header?

Reply:
I am still hesitating to allow any header name though the api in 
c8158f0 is simpler.  The only downside is that user need to spell the 
HOST correctly in the format of ":authority". 

Does anyone need a flexible header name?

Reply:
legacy `host` should also be OK. I just think it's better to do it once and for all since we're already extending and enhancing the API.

Reply:
+1

Reply:
I think the conclusion is that we support header names other than "x-envoy-original-dst-host" and ":authority"

Reply:
I would just allow the user to set any header and have the default be what it is today? They can set `:authority` or whatever and you can document it as such? This is already done elsewhere for similar cases.

Reply:
LGTM. That will be as single as a `string header_name = 2`. Updating...

Reply:
Update api in https://github.com/envoyproxy/envoy/pull/20534/commits/613a00791f8724a3e425ddc9307bda9427700355